tutorials/envoy-flask-google-container-engine.md

```diff
@@ -0,0 +1,566 @@
+---
+title: Deploying Envoy with a Python Flask webapp and Google Container Engine
+description: Learn how to use Envoy in Google Container Engine as a foundation for adding resilience and observability to a microservice-based application.
+author: flynn
+tags: microservices, Container Engine, Envoy, Flask, Python
+date_published: 2017-06-19
+---
+
+One of the recurring problems with using microservices is managing communications. Your clients need to be able to speak to your services, and in most cases services need to speak amongst themselves. When things go wrong, the system as a whole needs to be resilient, so that it degrades gracefully instead of catastrophically, but it also needs to be observable so that you can figure out what's wrong.
+
+A useful pattern is to enlist a proxy, like [Envoy](https://lyft.github.io/envoy/) to help [make your application more resilient and observable](https://www.datawire.io/guide/traffic/getting-started-lyft-envoy-microservices-resilience/). Envoy can be a bit daunting to set up, so this tutorial will walk you through deploying a Python Flask webapp with Envoy on Google Container Engine.
+
+## The Application
+
+Our application is a super simple REST-based user service: it can create users, read information about a user, and process simple logins. Even a trivial application like this involves several real-world concerns, though:
+
+* It requires persistent storage.
+* It will let us explore scaling the different pieces of the application.
+* It will let us explore Envoy at the edge, where the user’s client talks to our application.
+* It will let us explore Envoy internally, brokering communications between the various parts of the application.
+
+Envoy runs as a sidecar, so it's language-agnostic. For this tutorial, we’ll use Python and Flask for simplicity. We’ll use PostgreSQL for persistence because it has good Python support and it’s easy to get running both locally and in the cloud. And of course, running on Google Container Engine means managing everything with Kubernetes.
+
+## Setting Up
+
+### Google Container Engine
+
+You'll need a Google Cloud Platform account for this, so that you can set up a Google Container Engine cluster, just visit the [Google Cloud Platform Console](https://console.cloud.google.com/kubernetes) and use the UI to create a new cluster. Picking the defaults should be fine for our purposes.
+
+### Kubernetes
+
+You'll need `kubectl`, the Kubernetes CLI, to work with Google Container Engine. On a Mac you can use `brew install kubernetes-cli`; else you'll need to follow the [Kubernetes installation intructions](https://kubernetes.io/docs/user-guide/prereqs/).
```

Comment:
I suggest update link to https://kubernetes.io/docs/tasks/tools/install-kubectl/ and replacing `;` with `,`.